---
title: "Performing scClassify using pretrained model"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{pretrainedModel}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


This vignette will provide an example showing how one can use a pretrained model to predict cell types.

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  comment = "#>"
)
```



# Setting up the data

We assume that you have *log-transformed* (size-factor normalized) matrices as query datasets, where each row refers to a gene and each column a cell. For demonstration purposes, we will take a subset of single-cell pancreas datasets from one independent study (Wang et al.).


```{r setup}
library(scClassify)
load("data/scClassify_example.rda")
wang_cellTypes <- scClassify_example$wang_cellTypes
exprsMat_wang_subset <- scClassify_example$exprsMat_wang_subset
exprsMat_wang_subset <- as(exprsMat_wang_subset, "dgCMatrix")
```

Here, we load our pretrained model using a subset of the Xin et al. human pancreas dataset as our reference data.

First, let us check basic information relating to our pretrained model. 

```{r}
load("data/trainClassExample.rda")
trainClassExample
```

In this pretrained model, we have selected the genes based on Differential Expression using limma. To check the genes that are available in the pretrained model: 


```{r}
trainClassExample@features
```


We can also visualise the cell type tree of the reference data.

```{r}
plotCellTypeTree(trainClassExample@cellTypeTree)
```

# Running scClassify

Next, we perform `predict_scClassify` with our pretrained model `trainRes = trainClassExample` to predict the cell types of our query data matrix `exprsMat_wang_subset_sparse`. Here, we used `pearson` and `spearman` as similarity metrics.

```{r}
pred_res <- predict_scClassify(exprsMat_test = exprsMat_wang_subset,
                               trainRes = trainClassExample,
                               cellTypes_test = wang_cellTypes,
                               algorithm = "WKNN",
                               features = c("limma"),
                               similarity = c("pearson", "spearman"),
                               prob_threshold = 0.7,
                               verbose = TRUE)
```

Noted that the `cellType_test` is not a required input. For datasets with unknown labels, users can simply leave it as `cellType_test = NULL`.



Prediction results for pearson as the similarity metric:


```{r}
table(pred_res$pearson_WKNN_limma$predRes, wang_cellTypes)
```

Prediction results for spearman as the similarity metric:

```{r}
table(pred_res$spearman_WKNN_limma$predRes, wang_cellTypes)
```



# Session Info

```{r}
sessionInfo()
```
