---
title: "scClassify Model Building and Prediction"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{scClassify}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Introduction

A common application of single-cell RNA sequencing (RNA-seq) data is to identify discrete cell types. To take advantage of the large collection of well-annotated scRNA-seq datasets, `scClassify` package implements a set of methods to perform accurate cell type classification based on *ensemble learning* and *sample size calculation*. This vignette demonstrates the usage of `scClassify`, providing a pithy description of each method with workable examples. 

# Setting up the data

We assume that you have *log-transformed* (size-factor normalized) matrices where each row is a gene and each column a cell for a reference dataset and a query dataset. For demonstration purposes, we will take a subset of single-cell pancreas datasets from two independent studies (Wang et al., and Xin et al.).


```{r setup}
library("scClassify")
load("data/scClassify_example.rda")
```




```{r}
xin_cellTypes <- scClassify_example$xin_cellTypes
exprsMat_xin_subset <- scClassify_example$exprsMat_xin_subset
wang_cellTypes <- scClassify_example$wang_cellTypes
exprsMat_wang_subset <- scClassify_example$exprsMat_wang_subset
exprsMat_xin_subset <- as(exprsMat_xin_subset, "dgCMatrix")
exprsMat_wang_subset <- as(exprsMat_wang_subset, "dgCMatrix")
```


The original cell type annotations and the example datasets can be easily accessed as shown below. The cell type compositions can be


```{r}
cat("Cell type composition of Xin et al. data")
table(xin_cellTypes)
cat("Cell type composition of Wang et al. data")
table(wang_cellTypes)
```

We can see that Xin et al. data only have 4 cell types, while Wang et al. has 7 cell types

# scClassify



## Non-ensemble scClassify

We first perform non-ensemble `scClassify` by using Xin et al. as reference dataset, and Wang et al data as query data, using `WKNN` as KNN algorithm, `DE` (differential expression genes) as gene selection method, and `pearson` as similarity calculation method.

```{r}
scClassify_res <- scClassify(exprsMat_train = as.matrix(exprsMat_xin_subset),
                      cellTypes_train = xin_cellTypes,
                      exprsMat_test = list(wang = as.matrix(exprsMat_wang_subset)),
                      cellTypes_test = list(wang = wang_cellTypes),
                      tree = "HOPACH",
                      algorithm = "WKNN",
                      selectFeatures = c("limma"),
                      similarity = c("pearson"),
                      returnList = FALSE,
                      verbose = FALSE)

```

Check the cell type tree generated by the reference data

```{r warning=FALSE}
scClassify_res$trainRes
plotCellTypeTree(scClassify_res$trainRes@cellTypeTree)
```

Noted that `scClassify_res$trainRes` is a `scClassifyTrainModel` class.


Check the prediction results.

```{r}
table(scClassify_res$testRes$wang$pearson_WKNN_limma$predRes, wang_cellTypes)
```




## Ensemble Classify

We next perform ensemble `scClassify` by using Xin et al. as reference dataset, and Wang et al data as query data, using `WKNN` as KNN algorithm, `DE` as gene selection method, and `pearson`, `spearman` as similarity calculation methods. Thus, we are performing two combinations of gene selection models and similarity metrics as train classifiers, that is, 

1. `WKNN` + `DE` + `pearson`
2. `WKNN` + `DE` + `spearman`

Here, we will weight this two classifiers equally by setting `wegihted_ensemble = FALSE`. By default this is set as `TRUE`, and each base classifiers will be weighted by the accuracy rates trained in reference data.

```{r}
scClassify_res_ensemble <- scClassify(exprsMat_train = as.matrix(exprsMat_xin_subset),
                                      cellTypes_train = xin_cellTypes,
                                      exprsMat_test = list(wang = as.matrix(exprsMat_wang_subset)),
                                      cellTypes_test = list(wang = wang_cellTypes),
                                      tree = "HOPACH",
                                      algorithm = "WKNN",
                                      selectFeatures = c("limma"),
                                      similarity = c("pearson", "cosine"),
                                      weighted_ensemble = FALSE,
                                      returnList = FALSE,
                                      verbose = FALSE)

```

Comparing two base classifiers prediction 

```{r}
table(scClassify_res_ensemble$testRes$wang$pearson_WKNN_limma$predRes,
      scClassify_res_ensemble$testRes$wang$cosine_WKNN_limma$predRes)
```

Check the final ensemble results

```{r}
table(scClassify_res_ensemble$testRes$wang$ensembleRes$cellTypes, 
      wang_cellTypes)
```

# Session Info

```{r}
sessionInfo()
```
