<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Performing scClassify using pretrained model â€¢ scClassify</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Performing scClassify using pretrained model">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scClassify</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Classify Your Cells
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pretrainedModel.html">Prediction using pretrained model</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Build Your Own Models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/scClassify.html">scClassify Model Building and Prediction</a>
    </li>
    <li>
      <a href="../articles/sampleSizeCal.html">Sample size learning</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Pretrained Models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/webOnly/Tasic2016_mouseNeuronal.html">Tasic (2016) (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Tasic2018_mouseNeuronal.html">Tasic (2018) (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Hrvatin_mouseNeuronal.html">Hrvatin (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Cohen_mouseLung.html">Cohen (Mouse lung)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Park_mouseKidney.html">Park (Mouse kidney)</a>
    </li>
    <li>
      <a href="../articles/webOnly/MacParlandres_humanLiver.html">MacParland (Human Liver)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Aizarani_humanLiver.html">Aizarani (Human Liver)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Baron_humanPancreas.html">Baron (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Muraro_humanPancreas.html">Muraro (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Lawlor_humanPancreas.html">Lawlor (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Xin_humanPancreas.html">Xin (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Wang_humanPancreas.html">Wang (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Segerstolpe_humanPancreas.html">Segerstolpe (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Joint_humanPancreas.html">Joint (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Li_humanMelanoma.html">Li (Human Melanoma)</a>
    </li>
    <li>
      <a href="../articles/webOnly/TabulaMuris.html">Tabula Muris (Mouse cell atlas)</a>
    </li>
  </ul>
</li>
<li>
  <a href="http://shiny.maths.usyd.edu.au/scClassify/">Interactive scClassify shiny app</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/SydneyBioX/scClassify">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Performing scClassify using pretrained model</h1>
                        <h4 class="author">Yingxin Lin</h4>
            
      
      
      <div class="hidden name"><code>pretrainedModel.Rmd</code></div>

    </div>

    
    
<p>This vignette will provide an example showing how to use pretrained model we provided to predict cell types.</p>
<div id="setting-up-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h1>
<p>We assume that you have <em>log-transformed</em> (size-factor normalized) matrices where each row is a gene and each column a cell as a query dataset. For demonstration purposes, we will take a subset of single-cell pancreas datasets from one independent studies (Wang et al.).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/load">load</a></span>(<span class="st">"data/scClassify_example.rda"</span>)
wang_cellTypes &lt;-<span class="st"> </span>scClassify_example<span class="op">$</span>wang_cellTypes
exprsMat_wang_subset &lt;-<span class="st"> </span>scClassify_example<span class="op">$</span>exprsMat_wang_subset
exprsMat_wang_subset &lt;-<span class="st"> </span><span class="kw">as</span>(exprsMat_wang_subset, <span class="st">"dgCMatrix"</span>)</code></pre></div>
<p>Here, we load our pretrained model using a subset of Xin et al. human pancreas as reference data.</p>
<p>First, check basic info of our pretrained model.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(scClassify)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/load">load</a></span>(<span class="st">"data/trainClassExample.rda"</span>)
trainClassExample
<span class="co">#&gt; Class: scClassifyTrainModel </span>
<span class="co">#&gt; Model name: training </span>
<span class="co">#&gt; Feature selection methods: limma </span>
<span class="co">#&gt; Number of cells in the training data: 674 </span>
<span class="co">#&gt; Number of cell types in the training data: 4</span></code></pre></div>
<p>In this pretrained model, we have select the genes based on Differential Expression using limma. To check the genes that are available in the pretrained model:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trainClassExample<span class="op">@</span>features
<span class="co">#&gt; [1] "limma"</span></code></pre></div>
<p>We can also visualise the cell type tree of the reference data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plotCellTypeTree.html">plotCellTypeTree</a></span>(trainClassExample<span class="op">@</span>cellTypeTree)</code></pre></div>
<p><img src="pretrainedModel_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div id="running-scclassify" class="section level1">
<h1 class="hasAnchor">
<a href="#running-scclassify" class="anchor"></a>Running scClassify</h1>
<p>Next, we perform <code>predict_scClassify</code> to predict the cell types of our query data matrix <code>exprsMat_wang_subset_sparse</code>, using our pretrained model <code>trainRes = trainClassExample</code>. Here, we used <code>pearson</code> and <code>spearman</code> as similarity metrics.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pred_res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/predict_scClassify.html">predict_scClassify</a></span>(<span class="dt">exprsMat_test =</span> exprsMat_wang_subset,
                               <span class="dt">trainRes =</span> trainClassExample,
                               <span class="dt">cellTypes_test =</span> wang_cellTypes,
                               <span class="dt">algorithm =</span> <span class="st">"WKNN"</span>,
                               <span class="dt">features =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"limma"</span>),
                               <span class="dt">similarity =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"pearson"</span>, <span class="st">"spearman"</span>),
                               <span class="dt">prob_threshold =</span> <span class="fl">0.7</span>,
                               <span class="dt">verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; Using parameters: </span>
<span class="co">#&gt; similarity  algorithm   features </span>
<span class="co">#&gt;  "pearson"     "WKNN"    "limma" </span>
<span class="co">#&gt; [1] "Using dynamic correlation cutoff..."</span>
<span class="co">#&gt; [1] "Using dynamic correlation cutoff..."</span>
<span class="co">#&gt; classify_res</span>
<span class="co">#&gt;                correct   correctly unassigned           intermediate </span>
<span class="co">#&gt;            0.704590818            0.249500998            0.000000000 </span>
<span class="co">#&gt; incorrectly unassigned         error assigned          misclassified </span>
<span class="co">#&gt;            0.000000000            0.041916168            0.003992016 </span>
<span class="co">#&gt; Using parameters: </span>
<span class="co">#&gt; similarity  algorithm   features </span>
<span class="co">#&gt; "spearman"     "WKNN"    "limma" </span>
<span class="co">#&gt; [1] "Using dynamic correlation cutoff..."</span>
<span class="co">#&gt; [1] "Using dynamic correlation cutoff..."</span>
<span class="co">#&gt; classify_res</span>
<span class="co">#&gt;                correct   correctly unassigned           intermediate </span>
<span class="co">#&gt;            0.702594810            0.013972056            0.000000000 </span>
<span class="co">#&gt; incorrectly unassigned         error assigned          misclassified </span>
<span class="co">#&gt;            0.001996008            0.277445110            0.003992016</span></code></pre></div>
<p>Noted that the <code>cellType_test</code> is not a necessarily input. For the dataset with unknown label, users can simply leave it as <code>cellType_test = NULL</code>.</p>
<p>Pearson as similarity metrics prediction results:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(pred_res<span class="op">$</span>pearson_WKNN_limma<span class="op">$</span>predRes, wang_cellTypes)
<span class="co">#&gt;                   wang_cellTypes</span>
<span class="co">#&gt;                    acinar alpha beta delta ductal gamma stellate</span>
<span class="co">#&gt;   alpha                 0   206    0     0      0     2        0</span>
<span class="co">#&gt;   beta                  0     0  118     0      1     0        0</span>
<span class="co">#&gt;   beta_delta_gamma      0     0    0     0     20     0        0</span>
<span class="co">#&gt;   delta                 0     0    0    10      0     0        0</span>
<span class="co">#&gt;   gamma                 0     0    0     0      0    19        0</span>
<span class="co">#&gt;   unassigned            5     0    0     0     75     0       45</span></code></pre></div>
<p>Spearman as similarity metrics prediction results:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(pred_res<span class="op">$</span>spearman_WKNN_limma<span class="op">$</span>predRes, wang_cellTypes)
<span class="co">#&gt;                   wang_cellTypes</span>
<span class="co">#&gt;                    acinar alpha beta delta ductal gamma stellate</span>
<span class="co">#&gt;   alpha                 0   206    0     0      0     2        2</span>
<span class="co">#&gt;   beta                  2     0  118     0     30     0        7</span>
<span class="co">#&gt;   beta_delta_gamma      1     0    0     0     65     0       30</span>
<span class="co">#&gt;   delta                 0     0    0    10      0     0        2</span>
<span class="co">#&gt;   gamma                 0     0    0     0      0    18        0</span>
<span class="co">#&gt;   unassigned            2     0    0     0      1     1        4</span></code></pre></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()
<span class="co">#&gt; R version 3.6.1 (2019-07-05)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Mojave 10.14.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] scClassify_0.2.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Biobase_2.44.0      viridis_0.5.1       mixtools_1.1.0     </span>
<span class="co">#&gt;  [4] viridisLite_0.3.0   splines_3.6.1       ggraph_1.0.2       </span>
<span class="co">#&gt;  [7] RcppParallel_4.4.3  assertthat_0.2.1    stats4_3.6.1       </span>
<span class="co">#&gt; [10] yaml_2.2.0          ggrepel_0.8.1       pillar_1.4.2       </span>
<span class="co">#&gt; [13] backports_1.1.4     lattice_0.20-38     glue_1.3.1         </span>
<span class="co">#&gt; [16] limma_3.40.6        digest_0.6.20       polyclip_1.10-0    </span>
<span class="co">#&gt; [19] colorspace_1.4-1    htmltools_0.3.6     Matrix_1.2-17      </span>
<span class="co">#&gt; [22] plyr_1.8.4          pkgconfig_2.0.2     purrr_0.3.2        </span>
<span class="co">#&gt; [25] scales_1.0.0        tweenr_1.0.1        hopach_2.44.0      </span>
<span class="co">#&gt; [28] ggforce_0.3.1       tibble_2.1.3        proxy_0.4-23       </span>
<span class="co">#&gt; [31] mgcv_1.8-28         farver_1.1.0        ggplot2_3.2.1      </span>
<span class="co">#&gt; [34] BiocGenerics_0.31.5 lazyeval_0.2.2      survival_2.44-1.1  </span>
<span class="co">#&gt; [37] magrittr_1.5        crayon_1.3.4        memoise_1.1.0      </span>
<span class="co">#&gt; [40] evaluate_0.14       fs_1.3.1            nlme_3.1-141       </span>
<span class="co">#&gt; [43] MASS_7.3-51.4       segmented_1.0-0     xml2_1.2.2         </span>
<span class="co">#&gt; [46] tools_3.6.1         minpack.lm_1.2-1    stringr_1.4.0      </span>
<span class="co">#&gt; [49] S4Vectors_0.23.21   munsell_0.5.0       cluster_2.1.0      </span>
<span class="co">#&gt; [52] compiler_3.6.1      pkgdown_1.3.0       proxyC_0.1.5       </span>
<span class="co">#&gt; [55] pbmcapply_1.5.0     rlang_0.4.0         grid_3.6.1         </span>
<span class="co">#&gt; [58] rstudioapi_0.10     igraph_1.2.4.1      labeling_0.3       </span>
<span class="co">#&gt; [61] rmarkdown_1.15      gtable_0.3.0        roxygen2_6.1.1     </span>
<span class="co">#&gt; [64] R6_2.4.0            gridExtra_2.3       knitr_1.24         </span>
<span class="co">#&gt; [67] dplyr_0.8.3         commonmark_1.7      rprojroot_1.3-2    </span>
<span class="co">#&gt; [70] desc_1.2.0          stringi_1.4.3       parallel_3.6.1     </span>
<span class="co">#&gt; [73] Rcpp_1.0.2          tidyselect_0.2.5    xfun_0.9           </span>
<span class="co">#&gt; [76] diptest_0.75-7</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#setting-up-the-data">Setting up the data</a></li>
      <li><a href="#running-scclassify">Running scClassify</a></li>
      <li><a href="#session-info">Session Info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Yingxin Lin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
