<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scClassify Model Building and Prediction • scClassify</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scClassify Model Building and Prediction">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scClassify</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Classify Your Cells
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pretrainedModel.html">Prediction using pretrained model</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Build Your Own Models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/scClassify.html">scClassify Model Building and Prediction</a>
    </li>
    <li>
      <a href="../articles/webOnly/sampleSizeCal.html">Sample size learning</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Pretrained Models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/webOnly/Tasic2016_mouseNeuronal.html">Tasic (2016) (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Tasic2018_mouseNeuronal.html">Tasic (2018) (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Hrvatin_mouseNeuronal.html">Hrvatin (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Cohen_mouseLung.html">Cohen (Mouse lung)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Park_mouseKidney.html">Park (Mouse kidney)</a>
    </li>
    <li>
      <a href="../articles/webOnly/MacParlandres_humanLiver.html">MacParland (Human Liver)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Aizarani_humanLiver.html">Aizarani (Human Liver)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Baron_humanPancreas.html">Baron (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Muraro_humanPancreas.html">Muraro (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Lawlor_humanPancreas.html">Lawlor (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Xin_humanPancreas.html">Xin (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Wang_humanPancreas.html">Wang (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Segerstolpe_humanPancreas.html">Segerstolpe (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Joint_humanPancreas.html">Joint (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Joint_humanPBMC.html">Joint (Human PBMC)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Li_humanMelanoma.html">Li (Human Melanoma)</a>
    </li>
    <li>
      <a href="../articles/webOnly/TabulaMuris.html">Tabula Muris (Mouse cell atlas)</a>
    </li>
  </ul>
</li>
<li>
  <a href="http://shiny.maths.usyd.edu.au/scClassify/">
    <span class="fa fa-star"></span>
     
    Interactive scClassify shiny app (beta)
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/SydneyBioX/scClassify">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>scClassify Model Building and Prediction</h1>
            
      
      
      <div class="hidden name"><code>scClassify.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>A common application of single-cell RNA sequencing (RNA-seq) data is to identify discrete cell types. To take advantage of the large collection of well-annotated scRNA-seq datasets, <code>scClassify</code> package implements a set of methods to perform accurate cell type classification based on <em>ensemble learning</em> and <em>sample size calculation</em>. This vignette demonstrates the usage of <code>scClassify</code>, providing a pithy description of each method with workable examples.</p>
</div>
<div id="setting-up-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h1>
<p>We assume that you have <em>log-transformed</em> (size-factor normalized) matrices where each row is a gene and each column a cell for a reference dataset and a query dataset. For demonstration purposes, we will take a subset of single-cell pancreas datasets from two independent studies (Wang et al., and Xin et al.).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(<span class="st">"scClassify"</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/load">load</a></span>(<span class="st">"data/scClassify_example.rda"</span>)
xin_cellTypes &lt;-<span class="st"> </span>scClassify_example<span class="op">$</span>xin_cellTypes
exprsMat_xin_subset &lt;-<span class="st"> </span>scClassify_example<span class="op">$</span>exprsMat_xin_subset
wang_cellTypes &lt;-<span class="st"> </span>scClassify_example<span class="op">$</span>wang_cellTypes
exprsMat_wang_subset &lt;-<span class="st"> </span>scClassify_example<span class="op">$</span>exprsMat_wang_subset
exprsMat_xin_subset &lt;-<span class="st"> </span><span class="kw">as</span>(exprsMat_xin_subset, <span class="st">"dgCMatrix"</span>)
exprsMat_wang_subset &lt;-<span class="st"> </span><span class="kw">as</span>(exprsMat_wang_subset, <span class="st">"dgCMatrix"</span>)</code></pre></div>
<p>The original cell type annotations and compositions of the example datasets can be easily accessed as shown below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat</a></span>(<span class="st">"Cell type composition of Xin et al. data"</span>)
<span class="co">#&gt; Cell type composition of Xin et al. data</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(xin_cellTypes)
<span class="co">#&gt; xin_cellTypes</span>
<span class="co">#&gt; alpha  beta delta gamma </span>
<span class="co">#&gt;   285   261    49    79</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/cat">cat</a></span>(<span class="st">"Cell type composition of Wang et al. data"</span>)
<span class="co">#&gt; Cell type composition of Wang et al. data</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(wang_cellTypes)
<span class="co">#&gt; wang_cellTypes</span>
<span class="co">#&gt;   acinar    alpha     beta    delta   ductal    gamma stellate </span>
<span class="co">#&gt;        5      206      118       10       96       21       45</span></code></pre></div>
<p>We can see that Xin et al. data only have 4 cell types, while Wang et al. has 7 cell types.</p>
</div>
<div id="scclassify" class="section level1">
<h1 class="hasAnchor">
<a href="#scclassify" class="anchor"></a>scClassify</h1>
<div id="non-ensemble-scclassify" class="section level2">
<h2 class="hasAnchor">
<a href="#non-ensemble-scclassify" class="anchor"></a>Non-ensemble scClassify</h2>
<p>We first perform non-ensemble <code>scClassify</code> by using Xin et al. as our reference dataset and Wang et al. data as ur query data. We use <code>WKNN</code> as the KNN algorithm, <code>DE</code> (differential expression genes) as the gene selection method, and lastly <code>pearson</code> as the similarity calculation method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scClassify_res &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scClassify.html">scClassify</a></span>(<span class="dt">exprsMat_train =</span> exprsMat_xin_subset,
                      <span class="dt">cellTypes_train =</span> xin_cellTypes,
                      <span class="dt">exprsMat_test =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">wang =</span> exprsMat_wang_subset),
                      <span class="dt">cellTypes_test =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">wang =</span> wang_cellTypes),
                      <span class="dt">tree =</span> <span class="st">"HOPACH"</span>,
                      <span class="dt">algorithm =</span> <span class="st">"WKNN"</span>,
                      <span class="dt">selectFeatures =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"limma"</span>),
                      <span class="dt">similarity =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"pearson"</span>),
                      <span class="dt">returnList =</span> <span class="ot">FALSE</span>,
                      <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>We can check the cell type tree generated by the reference data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scClassify_res<span class="op">$</span>trainRes
<span class="co">#&gt; Class: scClassifyTrainModel </span>
<span class="co">#&gt; Model name: training </span>
<span class="co">#&gt; Feature selection methods: limma </span>
<span class="co">#&gt; Number of cells in the training data: 674 </span>
<span class="co">#&gt; Number of cell types in the training data: 4</span>
<span class="kw"><a href="../reference/plotCellTypeTree.html">plotCellTypeTree</a></span>(scClassify_res<span class="op">$</span>trainRes<span class="op">@</span>cellTypeTree)</code></pre></div>
<p><img src="scClassify_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>Noted that <code>scClassify_res$trainRes</code> is a <code>scClassifyTrainModel</code> class.</p>
<p>Check the prediction results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(scClassify_res<span class="op">$</span>testRes<span class="op">$</span>wang<span class="op">$</span>pearson_WKNN_limma<span class="op">$</span>predRes, wang_cellTypes)
<span class="co">#&gt;                   wang_cellTypes</span>
<span class="co">#&gt;                    acinar alpha beta delta ductal gamma stellate</span>
<span class="co">#&gt;   alpha                 0   206    0     0      0     2        0</span>
<span class="co">#&gt;   beta                  0     0  118     0      1     0        0</span>
<span class="co">#&gt;   beta_delta_gamma      0     0    0     0     20     0        0</span>
<span class="co">#&gt;   delta                 0     0    0    10      0     0        0</span>
<span class="co">#&gt;   gamma                 0     0    0     0      0    19        0</span>
<span class="co">#&gt;   unassigned            5     0    0     0     75     0       45</span></code></pre></div>
</div>
<div id="ensemble-classify" class="section level2">
<h2 class="hasAnchor">
<a href="#ensemble-classify" class="anchor"></a>Ensemble Classify</h2>
<p>We next perform ensemble <code>scClassify</code> by using Xin et al. as our reference dataset and Wang et al. data as our query data. We use <code>WKNN</code> as the KNN algorithm, <code>DE</code> as the gene selection method, and <code>pearson</code> and <code>spearman</code> as the similarity calculation methods. Thus, we will generate two combinations of gene selection models and similarity metrics as training classifiers:</p>
<ol style="list-style-type: decimal">
<li>
<code>WKNN</code> + <code>DE</code> + <code>pearson</code>
</li>
<li>
<code>WKNN</code> + <code>DE</code> + <code>spearman</code>
</li>
</ol>
<p>Here, we will weight these two classifiers equally by setting <code>wegihted_ensemble = FALSE</code>. By default this is set as <code>TRUE</code>, so each base classifier will be weighted by the accuracy rates trained in the reference data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scClassify_res_ensemble &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scClassify.html">scClassify</a></span>(<span class="dt">exprsMat_train =</span> exprsMat_xin_subset,
                                      <span class="dt">cellTypes_train =</span> xin_cellTypes,
                                      <span class="dt">exprsMat_test =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">wang =</span> exprsMat_wang_subset),
                                      <span class="dt">cellTypes_test =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">wang =</span> wang_cellTypes),
                                      <span class="dt">tree =</span> <span class="st">"HOPACH"</span>,
                                      <span class="dt">algorithm =</span> <span class="st">"WKNN"</span>,
                                      <span class="dt">selectFeatures =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"limma"</span>),
                                      <span class="dt">similarity =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"pearson"</span>, <span class="st">"cosine"</span>),
                                      <span class="dt">weighted_ensemble =</span> <span class="ot">FALSE</span>,
                                      <span class="dt">returnList =</span> <span class="ot">FALSE</span>,
                                      <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>We can compare the two base classifiers predictions as below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(scClassify_res_ensemble<span class="op">$</span>testRes<span class="op">$</span>wang<span class="op">$</span>pearson_WKNN_limma<span class="op">$</span>predRes,
      scClassify_res_ensemble<span class="op">$</span>testRes<span class="op">$</span>wang<span class="op">$</span>cosine_WKNN_limma<span class="op">$</span>predRes)
<span class="co">#&gt;                   </span>
<span class="co">#&gt;                    alpha beta beta_delta_gamma delta gamma unassigned</span>
<span class="co">#&gt;   alpha              208    0                0     0     0          0</span>
<span class="co">#&gt;   beta                 0  119                0     0     0          0</span>
<span class="co">#&gt;   beta_delta_gamma     0    0               13     0     0          7</span>
<span class="co">#&gt;   delta                0    0                0    10     0          0</span>
<span class="co">#&gt;   gamma                0    0                0     0    19          0</span>
<span class="co">#&gt;   unassigned           1    3                8     0     0        113</span></code></pre></div>
<p>Now, check the final ensemble results:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/table">table</a></span>(scClassify_res_ensemble<span class="op">$</span>testRes<span class="op">$</span>wang<span class="op">$</span>ensembleRes<span class="op">$</span>cellTypes, 
      wang_cellTypes)
<span class="co">#&gt;                   wang_cellTypes</span>
<span class="co">#&gt;                    acinar alpha beta delta ductal gamma stellate</span>
<span class="co">#&gt;   alpha                 0   206    0     0      0     2        0</span>
<span class="co">#&gt;   unassigned            5     0    0     0     75     0       45</span>
<span class="co">#&gt;   delta                 0     0    0    10      0     0        0</span>
<span class="co">#&gt;   beta_delta_gamma      0     0    0     0     20     0        0</span>
<span class="co">#&gt;   beta                  0     0  118     0      1     0        0</span>
<span class="co">#&gt;   gamma                 0     0    0     0      0    19        0</span></code></pre></div>
</div>
</div>
<div id="train-your-own-model" class="section level1">
<h1 class="hasAnchor">
<a href="#train-your-own-model" class="anchor"></a>Train your own model</h1>
<p>You can also train your own model <code>scClassifyTrainModel</code> using <code><a href="../reference/train_scClassify.html">train_scClassify()</a></code>. Note that by setting <code>weightsCal = TRUE</code>, we will calculate the training error of the reference data as the weights for the individual classifiers.</p>
<p>Here, we illustrate the training function with gene selection methods based on differential expression (“limma”) and biomodal distribution (“BI”).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trainClass &lt;-<span class="st"> </span><span class="kw"><a href="../reference/train_scClassify.html">train_scClassify</a></span>(<span class="dt">exprsMat_train =</span> exprsMat_xin_subset,
                 <span class="dt">cellTypes_train =</span> xin_cellTypes,
                 <span class="dt">selectFeatures =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"limma"</span>, <span class="st">"BI"</span>),
                 <span class="dt">returnList =</span> <span class="ot">FALSE</span>
                 )
<span class="co">#&gt; after filtering not expressed genes </span>
<span class="co">#&gt; [1] 980 674</span>
<span class="co">#&gt; [1] "Feature Selection..."</span>
<span class="co">#&gt; [1] "Number of genes selected to construct HOPACH tree 160"</span>
<span class="co">#&gt; [1] "Constructing tree ..."</span>
<span class="co">#&gt; [1] "Training...."</span>
<span class="co">#&gt; [1] "============ selecting features by: limma ============"</span>
<span class="co">#&gt; [1] "============ selecting features by: BI ============"</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trainClass
<span class="co">#&gt; Class: scClassifyTrainModel </span>
<span class="co">#&gt; Model name: training </span>
<span class="co">#&gt; Feature selection methods: limma BI </span>
<span class="co">#&gt; Number of cells in the training data: 674 </span>
<span class="co">#&gt; Number of cell types in the training data: 4</span></code></pre></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()
<span class="co">#&gt; R version 3.6.1 (2019-07-05)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Mojave 10.14.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] scClassify_0.2.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Biobase_2.44.0      viridis_0.5.1       mixtools_1.1.0     </span>
<span class="co">#&gt;  [4] viridisLite_0.3.0   splines_3.6.1       ggraph_1.0.2       </span>
<span class="co">#&gt;  [7] RcppParallel_4.4.3  assertthat_0.2.1    statmod_1.4.32     </span>
<span class="co">#&gt; [10] stats4_3.6.1        yaml_2.2.0          ggrepel_0.8.1      </span>
<span class="co">#&gt; [13] pillar_1.4.2        backports_1.1.4     lattice_0.20-38    </span>
<span class="co">#&gt; [16] glue_1.3.1          limma_3.40.6        digest_0.6.20      </span>
<span class="co">#&gt; [19] polyclip_1.10-0     colorspace_1.4-1    htmltools_0.3.6    </span>
<span class="co">#&gt; [22] Matrix_1.2-17       plyr_1.8.4          pkgconfig_2.0.2    </span>
<span class="co">#&gt; [25] purrr_0.3.2         scales_1.0.0        tweenr_1.0.1       </span>
<span class="co">#&gt; [28] hopach_2.44.0       ggforce_0.3.1       tibble_2.1.3       </span>
<span class="co">#&gt; [31] proxy_0.4-23        mgcv_1.8-28         farver_1.1.0       </span>
<span class="co">#&gt; [34] ggplot2_3.2.1       BiocGenerics_0.31.5 lazyeval_0.2.2     </span>
<span class="co">#&gt; [37] survival_2.44-1.1   magrittr_1.5        crayon_1.3.4       </span>
<span class="co">#&gt; [40] memoise_1.1.0       evaluate_0.14       fs_1.3.1           </span>
<span class="co">#&gt; [43] nlme_3.1-141        MASS_7.3-51.4       segmented_1.0-0    </span>
<span class="co">#&gt; [46] xml2_1.2.2          tools_3.6.1         minpack.lm_1.2-1   </span>
<span class="co">#&gt; [49] stringr_1.4.0       S4Vectors_0.23.21   munsell_0.5.0      </span>
<span class="co">#&gt; [52] cluster_2.1.0       compiler_3.6.1      pkgdown_1.3.0      </span>
<span class="co">#&gt; [55] proxyC_0.1.5        pbmcapply_1.5.0     rlang_0.4.0        </span>
<span class="co">#&gt; [58] grid_3.6.1          rstudioapi_0.10     igraph_1.2.4.1     </span>
<span class="co">#&gt; [61] labeling_0.3        rmarkdown_1.15      gtable_0.3.0       </span>
<span class="co">#&gt; [64] roxygen2_6.1.1      R6_2.4.0            gridExtra_2.3      </span>
<span class="co">#&gt; [67] knitr_1.24          dplyr_0.8.3         commonmark_1.7     </span>
<span class="co">#&gt; [70] rprojroot_1.3-2     desc_1.2.0          stringi_1.4.3      </span>
<span class="co">#&gt; [73] parallel_3.6.1      Rcpp_1.0.2          tidyselect_0.2.5   </span>
<span class="co">#&gt; [76] xfun_0.9            diptest_0.75-7</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#setting-up-the-data">Setting up the data</a></li>
      <li>
<a href="#scclassify">scClassify</a><ul class="nav nav-pills nav-stacked">
<li><a href="#non-ensemble-scclassify">Non-ensemble scClassify</a></li>
      <li><a href="#ensemble-classify">Ensemble Classify</a></li>
      </ul>
</li>
      <li><a href="#train-your-own-model">Train your own model</a></li>
      <li><a href="#session-info">Session Info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Yingxin Lin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
