<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Performing scClassify using pretrained model â€¢ scClassify</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Performing scClassify using pretrained model">
<meta property="og:description" content="scClassify">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scClassify</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Classify Your Cells
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pretrainedModel.html">Prediction using pretrained model</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Build Your Own Models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/scClassify.html">scClassify Model Building and Prediction</a>
    </li>
    <li>
      <a href="../articles/webOnly/sampleSizeCal.html">Sample size learning</a>
    </li>
    <li>
      <a href="../articles/webOnly/scClassify_TM_heart.html">Case study - Tabula Muris Heart</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Pretrained Models
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/webOnly/Tasic2016_mouseNeuronal.html">Tasic (2016) (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Tasic2018_mouseNeuronal.html">Tasic (2018) (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Hrvatin_mouseNeuronal.html">Hrvatin (Mouse neuronal)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Cohen_mouseLung.html">Cohen (Mouse lung)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Park_mouseKidney.html">Park (Mouse kidney)</a>
    </li>
    <li>
      <a href="../articles/webOnly/MacParlandres_humanLiver.html">MacParland (Human Liver)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Aizarani_humanLiver.html">Aizarani (Human Liver)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Baron_humanPancreas.html">Baron (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Muraro_humanPancreas.html">Muraro (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Lawlor_humanPancreas.html">Lawlor (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Xin_humanPancreas.html">Xin (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Wang_humanPancreas.html">Wang (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Segerstolpe_humanPancreas.html">Segerstolpe (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Joint_humanPancreas.html">Joint (Human Pancreas)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Joint_humanPBMC.html">Joint (Human PBMC)</a>
    </li>
    <li>
      <a href="../articles/webOnly/Li_humanMelanoma.html">Li (Human Melanoma)</a>
    </li>
    <li>
      <a href="../articles/webOnly/TabulaMuris.html">Tabula Muris (Mouse cell atlas)</a>
    </li>
  </ul>
</li>
<li>
  <a href="http://shiny.maths.usyd.edu.au/scClassify/">
    <span class="fa fa-star"></span>
     
    Interactive scClassify shiny app (beta)
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/SydneyBioX/scClassify">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Performing scClassify using pretrained model</h1>
                        <h4 class="author">Yingxin Lin</h4>
            <address class="author_afil">
      School of Mathematics and Statistics, The University of Sydney, Australia<br><h4 class="date">20 April 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SydneyBioX/scClassify/blob/master/vignettes/pretrainedModel.Rmd"><code>vignettes/pretrainedModel.Rmd</code></a></small>
      <div class="hidden name"><code>pretrainedModel.Rmd</code></div>

    </address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>A common application of single-cell RNA sequencing (RNA-seq) data is to identify discrete cell types. To take advantage of the large collection of well-annotated scRNA-seq datasets, <code>scClassify</code> package implements a set of methods to perform accurate cell type classification based on <em>ensemble learning</em> and <em>sample size calculation</em>.</p>
<p>This vignette will provide an example showing how users can use a pretrained model of scClassify to predict cell types. A pretrained model is a <code>scClassifyTrainModel</code> object returned by <code><a href="https://rdrr.io/pkg/scClassify/man/train_scClassify.html">train_scClassify()</a></code>. A list of pretrained model can be found in <a href="https://sydneybiox.github.io/scClassify/index.html" class="uri">https://sydneybiox.github.io/scClassify/index.html</a>.</p>
<p>First, install <code>scClassify</code>, install <code>BiocManager</code> and use <code><a href="https://rdrr.io/pkg/BiocManager/man/install.html">BiocManager::install</a></code> to install <code>scClassify</code> package.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co"># installation of scClassify</span>
<span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>)) {
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
}
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"scClassify"</span>)</pre></body></html></div>
</div>
<div id="setting-up-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h1>
<p>We assume that you have <em>log-transformed</em> (size-factor normalized) matrices as query datasets, where each row refers to a gene and each column a cell. For demonstration purposes, we will take a subset of single-cell pancreas datasets from one independent study (Wang et al.).</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">scClassify</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"scClassify_example"</span>)
<span class="no">wang_cellTypes</span> <span class="kw">&lt;-</span> <span class="no">scClassify_example</span>$<span class="no">wang_cellTypes</span>
<span class="no">exprsMat_wang_subset</span> <span class="kw">&lt;-</span> <span class="no">scClassify_example</span>$<span class="no">exprsMat_wang_subset</span>
<span class="no">exprsMat_wang_subset</span> <span class="kw">&lt;-</span> <span class="fu">as</span>(<span class="no">exprsMat_wang_subset</span>, <span class="st">"dgCMatrix"</span>)</pre></body></html></div>
<p>Here, we load our pretrained model using a subset of the Xin et al. human pancreas dataset as our reference data.</p>
<p>First, let us check basic information relating to our pretrained model.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"trainClassExample_xin"</span>)
<span class="no">trainClassExample_xin</span>
<span class="co">#&gt; Class: scClassifyTrainModel </span>
<span class="co">#&gt; Model name: training </span>
<span class="co">#&gt; Feature selection methods: limma </span>
<span class="co">#&gt; Number of cells in the training data: 674 </span>
<span class="co">#&gt; Number of cell types in the training data: 4</span></pre></body></html></div>
<p>In this pretrained model, we have selected the genes based on Differential Expression using limma. To check the genes that are available in the pretrained model:</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/scClassify/man/features.html">features</a></span>(<span class="no">trainClassExample_xin</span>)
<span class="co">#&gt; [1] "limma"</span></pre></body></html></div>
<p>We can also visualise the cell type tree of the reference data.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/scClassify/man/plotCellTypeTree.html">plotCellTypeTree</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/scClassify/man/cellTypeTree.html">cellTypeTree</a></span>(<span class="no">trainClassExample_xin</span>))</pre></body></html></div>
<p><img src="pretrainedModel_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div id="running-scclassify" class="section level1">
<h1 class="hasAnchor">
<a href="#running-scclassify" class="anchor"></a>Running scClassify</h1>
<p>Next, we perform <code>predict_scClassify</code> with our pretrained model <code>trainRes = trainClassExample</code> to predict the cell types of our query data matrix <code>exprsMat_wang_subset_sparse</code>. Here, we used <code>pearson</code> and <code>spearman</code> as similarity metrics.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">pred_res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scClassify/man/predict_scClassify.html">predict_scClassify</a></span>(<span class="kw">exprsMat_test</span> <span class="kw">=</span> <span class="no">exprsMat_wang_subset</span>,
                               <span class="kw">trainRes</span> <span class="kw">=</span> <span class="no">trainClassExample_xin</span>,
                               <span class="kw">cellTypes_test</span> <span class="kw">=</span> <span class="no">wang_cellTypes</span>,
                               <span class="kw">algorithm</span> <span class="kw">=</span> <span class="st">"WKNN"</span>,
                               <span class="kw">features</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"limma"</span>),
                               <span class="kw">similarity</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"pearson"</span>, <span class="st">"spearman"</span>),
                               <span class="kw">prob_threshold</span> <span class="kw">=</span> <span class="fl">0.7</span>,
                               <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; Performing unweighted ensemble learning... </span>
<span class="co">#&gt; Using parameters: </span>
<span class="co">#&gt; similarity  algorithm   features </span>
<span class="co">#&gt;  "pearson"     "WKNN"    "limma" </span>
<span class="co">#&gt; [1] "Using dynamic correlation cutoff..."</span>
<span class="co">#&gt; [1] "Using dynamic correlation cutoff..."</span>
<span class="co">#&gt; classify_res</span>
<span class="co">#&gt;                correct   correctly unassigned           intermediate </span>
<span class="co">#&gt;            0.704590818            0.239520958            0.000000000 </span>
<span class="co">#&gt; incorrectly unassigned         error assigned          misclassified </span>
<span class="co">#&gt;            0.000000000            0.051896208            0.003992016 </span>
<span class="co">#&gt; Using parameters: </span>
<span class="co">#&gt; similarity  algorithm   features </span>
<span class="co">#&gt; "spearman"     "WKNN"    "limma" </span>
<span class="co">#&gt; [1] "Using dynamic correlation cutoff..."</span>
<span class="co">#&gt; [1] "Using dynamic correlation cutoff..."</span>
<span class="co">#&gt; classify_res</span>
<span class="co">#&gt;                correct   correctly unassigned           intermediate </span>
<span class="co">#&gt;            0.702594810            0.013972056            0.000000000 </span>
<span class="co">#&gt; incorrectly unassigned         error assigned          misclassified </span>
<span class="co">#&gt;            0.001996008            0.277445110            0.003992016 </span>
<span class="co">#&gt; weights for each base method: </span>
<span class="co">#&gt; numeric(0)</span></pre></body></html></div>
<p>Noted that the <code>cellType_test</code> is not a required input. For datasets with unknown labels, users can simply leave it as <code>cellType_test = NULL</code>.</p>
<p>Prediction results for pearson as the similarity metric:</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">pred_res</span>$<span class="no">pearson_WKNN_limma</span>$<span class="no">predRes</span>, <span class="no">wang_cellTypes</span>)
<span class="co">#&gt;                   wang_cellTypes</span>
<span class="co">#&gt;                    acinar alpha beta delta ductal gamma stellate</span>
<span class="co">#&gt;   alpha                 0   206    0     0      0     2        0</span>
<span class="co">#&gt;   beta                  0     0  118     0      1     0        0</span>
<span class="co">#&gt;   beta_delta_gamma      0     0    0     0     25     0        0</span>
<span class="co">#&gt;   delta                 0     0    0    10      0     0        0</span>
<span class="co">#&gt;   gamma                 0     0    0     0      0    19        0</span>
<span class="co">#&gt;   unassigned            5     0    0     0     70     0       45</span></pre></body></html></div>
<p>Prediction results for spearman as the similarity metric:</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">pred_res</span>$<span class="no">spearman_WKNN_limma</span>$<span class="no">predRes</span>, <span class="no">wang_cellTypes</span>)
<span class="co">#&gt;                   wang_cellTypes</span>
<span class="co">#&gt;                    acinar alpha beta delta ductal gamma stellate</span>
<span class="co">#&gt;   alpha                 0   206    0     0      0     2        2</span>
<span class="co">#&gt;   beta                  2     0  118     0     29     0        6</span>
<span class="co">#&gt;   beta_delta_gamma      1     0    0     0     66     0       31</span>
<span class="co">#&gt;   delta                 0     0    0    10      0     0        2</span>
<span class="co">#&gt;   gamma                 0     0    0     0      0    18        0</span>
<span class="co">#&gt;   unassigned            2     0    0     0      1     1        4</span></pre></body></html></div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R Under development (unstable) (2020-04-20 r78257)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 14.04.5 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so</span>
<span class="co">#&gt; LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] scClassify_0.99.4 BiocStyle_2.15.6 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Biobase_2.47.3      viridis_0.5.1       mixtools_1.2.0     </span>
<span class="co">#&gt;  [4] tidyr_1.0.2         tidygraph_1.1.2     viridisLite_0.3.0  </span>
<span class="co">#&gt;  [7] splines_4.1.0       ggraph_2.0.2        RcppParallel_5.0.0 </span>
<span class="co">#&gt; [10] assertthat_0.2.1    statmod_1.4.34      BiocManager_1.30.10</span>
<span class="co">#&gt; [13] stats4_4.1.0        yaml_2.2.1          ggrepel_0.8.2      </span>
<span class="co">#&gt; [16] pillar_1.4.3        backports_1.1.6     lattice_0.20-41    </span>
<span class="co">#&gt; [19] glue_1.4.0          limma_3.43.8        digest_0.6.25      </span>
<span class="co">#&gt; [22] polyclip_1.10-0     colorspace_1.4-1    htmltools_0.4.0    </span>
<span class="co">#&gt; [25] Matrix_1.2-18       pkgconfig_2.0.3     bookdown_0.18      </span>
<span class="co">#&gt; [28] purrr_0.3.4         scales_1.1.0        tweenr_1.0.1       </span>
<span class="co">#&gt; [31] hopach_2.47.0       BiocParallel_1.21.2 ggforce_0.3.1      </span>
<span class="co">#&gt; [34] tibble_3.0.1        proxy_0.4-23        mgcv_1.8-31        </span>
<span class="co">#&gt; [37] farver_2.0.3        ggplot2_3.3.0       ellipsis_0.3.0     </span>
<span class="co">#&gt; [40] BiocGenerics_0.33.3 survival_3.1-12     magrittr_1.5       </span>
<span class="co">#&gt; [43] crayon_1.3.4        memoise_1.1.0       evaluate_0.14      </span>
<span class="co">#&gt; [46] fs_1.4.1            nlme_3.1-147        MASS_7.3-51.5      </span>
<span class="co">#&gt; [49] segmented_1.1-0     tools_4.1.0         minpack.lm_1.2-1   </span>
<span class="co">#&gt; [52] lifecycle_0.2.0     stringr_1.4.0       S4Vectors_0.25.15  </span>
<span class="co">#&gt; [55] kernlab_0.9-29      munsell_0.5.0       cluster_2.1.0      </span>
<span class="co">#&gt; [58] compiler_4.1.0      pkgdown_1.5.1       proxyC_0.1.5       </span>
<span class="co">#&gt; [61] rlang_0.4.5         grid_4.1.0          igraph_1.2.5       </span>
<span class="co">#&gt; [64] labeling_0.3        rmarkdown_2.1       gtable_0.3.0       </span>
<span class="co">#&gt; [67] graphlayouts_0.6.0  R6_2.4.1            gridExtra_2.3      </span>
<span class="co">#&gt; [70] knitr_1.28          dplyr_0.8.5         rprojroot_1.3-2    </span>
<span class="co">#&gt; [73] desc_1.2.0          stringi_1.4.6       parallel_4.1.0     </span>
<span class="co">#&gt; [76] Rcpp_1.0.4.6        vctrs_0.2.4         diptest_0.75-7     </span>
<span class="co">#&gt; [79] tidyselect_1.0.0    xfun_0.13</span></pre></body></html></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yingxin Lin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
